http/client.go

```diff
@@ -8,114 +8,174 @@ import (
 	"errors"
 	"io/ioutil"
 	"net/http"
-	"os"
 	"path/filepath"
 	"strings"
 	"time"
 )
 
-// New creates a new http.Client with a custom certificate, which can be loaded from the passed CA Bundle file and/or
+var (
+	// ErrInvalidTransportType defines the error to be returned if the transport defined for the client is incorrect
+	ErrInvalidTransportType = errors.New("roundTripper transport invalid, should be http type")
+	// ErrEmptyCABundleFile defines the error to be returned if the WithCABundleFile option is called with an empty value
+	ErrEmptyCABundleFile = errors.New("caBundleFile can't be empty")
+	// ErrEmptyCABundleDir defines the error to be returned if the WithCABundleDir option is called with an empty value
+	ErrEmptyCABundleDir = errors.New("caBundleDir can't be empty")
+	// ErrEmptyAcceptInvalidHostname defines the error to be returned if the WithAcceptInvalidHostname option is called with an empty value
+	ErrEmptyAcceptInvalidHostname = errors.New("acceptInvalidHostname can't be empty")
```

Comment:
We could unify those into a single error:

`ErrEmptyArg = errors.New("argument is empty")`

And then decorate it on each function

```go
		if CABundleFile == "" {
			return fmt.Errorf("applying CABundleFile: %w", ErrEmptyArg)
		}
```

Reply:
Done